python/raf/_tvm_op/transform.py

```diff
@@ -419,3 +419,16 @@ def embedding_dx_compute(attrs, inputs, output_type):
 _reg.register_injective_schedule("raf.op.tvm.embedding_dx")
 
 _reg.register_strategy("raf.op.tvm.cumsum", strategy.cumsum_strategy)
+
+
+@register_compute("raf.op.tvm.group_cast")
+def group_cast_compute(attrs, inputs, output_type):
```

Comment:
This looks like just
```
let %1 = cast(%a, dtype);
let %2 = cast(%b, dtype);
let %3 = cast(%c, dtype);
let %4 = (%1, %2, %3);
%4
```
Seems like we could just use a pass to fuse them (e.g., CombineParallelCast) instead of introducing a new op.

Reply:
I was also thinking about making it more general and systematic. Whether we need to group the same ops together is complicated. Grouping decrease the overhead, but it delays the kernel issue time, which may increase the latency due to other ops are waiting for the result.  For collectives, I was also thinking about group different kind of op. If so we should pay more attention to the dependency. There is no motivation to do so for now.
If we want to make it systematic, we need to annotate each op that could be grouped and then implement a pass to analyze the whole graph to perform grouping. 
 
In ZeRO, we have the motivation to group some op together,  we know we can group the `cast` before `allgather` and the corresponding `allgahter` ops, this will not add extra dependency. For other `cast`, we do not group them in the graph. Implementing group ops is an easy way to achieve good performance without much engineering effort.  In my own option, I prefer to have this implementation for now and revisit this if we have a motivation to group other ops in other scenarios.  I think we also need to investigate the benefits of grouping for other scenarios.

Reply:
I agree that we should deliver the effective part ASAP.

While grouping different ops is definitely a bonus that we should revisit in the future, for grouping the same ops, IIUC, even we have a group op, you still need a pass to match a group of ops and replace them with a single group op, right? If so, I didn't see a problem of improving the current op to optionally take a list of tensors.

Reply:
Yes, we still need a pass. Let me improve the op and the pass.